router.go

```diff
@@ -116,13 +123,24 @@ func (r *Router) insert(method, path string, h HandlerFunc, t kind, ppath string
 		if sl < max {
 			max = sl
 		}
-		for ; l < max && search[l] == cn.prefix[l]; l++ {
+		if !r.echo.EnableCaseInsensitive {
+			for ; l < max && search[l] == cn.prefix[l]; l++ {
+			}
+		} else {
+			for ; l < max && (search[l] == cn.prefixL[l] || search[l] == cn.prefixU[l]); l++ {
+			}
 		}
 
 		if l == 0 {
 			// At root node
 			cn.label = search[0]
 			cn.prefix = search
+			if r.echo.EnableCaseInsensitive {
+				cn.prefixL = strings.ToLower(search)
```

Comment:
Only the lower-case variant is required. The lookup should do use the lower-caseed path then for the lookup.
This will save a bit of memory and cpu cycles. 

prefixLower should be used instead of prefixL and prefixU (same for label).